skbio/io/_base.py

```diff
@@ -195,7 +196,17 @@ def _format_fasta_like_records(generator, id_whitespace_replacement,
         qual = None
         if 'quality' in seq.positional_metadata:
             qual = seq.positional_metadata['quality'].values
-        yield header, str(seq), qual
+
+        if lowercase is not None:
```

Comment:
The semantics here should match the constructor, that means that either: `seq.lowercase` has to understand an inordinate amount of possible "keys" OR we restrict the constructor to be `str` or `bool`.  I would prefer the latter. @jairideout thoughts?